axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceTermController.java

```diff
@@ -119,25 +121,22 @@ public void initInvoiceTerm(ActionRequest request, ActionResponse response) {
       InvoiceTerm invoiceTerm = request.getContext().asType(InvoiceTerm.class);
       InvoiceTermService invoiceTermService = Beans.get(InvoiceTermService.class);
       Invoice invoice = null;
-      MoveLine moveLine = null;
+      MoveLine moveLine = ContextTool.getContextParent(request.getContext(), MoveLine.class, 1);
 
       if (request.getContext().getParent() != null) {
         invoice = ContextTool.getContextParent(request.getContext(), Invoice.class, 1);
-        if (invoice != null) {
-          invoiceTermService.initCustomizedInvoiceTerm(invoice, invoiceTerm);
-          response.setValues(invoiceTerm);
-        } else {
-          moveLine = ContextTool.getContextParent(request.getContext(), MoveLine.class, 1);
-
-          if (moveLine != null) {
-            Move move = ContextTool.getContextParent(request.getContext(), Move.class, 2);
-            invoiceTermService.initCustomizedInvoiceTerm(moveLine, invoiceTerm, move);
+      } else if (request.getContext().get("_invoiceId") != null) {
+        invoice =
+            Beans.get(InvoiceRepository.class)
+                .find(Long.valueOf((Integer) request.getContext().get("_invoiceId")));
+      }
 
-            if (move != null) {
-              moveLine.setMove(move);
-            }
-          }
-        }
+      if (invoice == null && moveLine != null) {
+        Move move = ContextTool.getContextParent(request.getContext(), Move.class, 2);
+        invoiceTermService.initCustomizedInvoiceTerm(moveLine, invoiceTerm, move);
+      } else if (invoice != null) {
+        invoiceTermService.initCustomizedInvoiceTerm(invoice, invoiceTerm);
+        response.setValues(invoiceTerm);
```

Comment:
I think you can remove the setValues() here, since it will always be called at the end

Reply:
Solved